### ISSUE TITLE
URGENT: Export/Import Page Section Code Not Working

### ISSUE THREAD

**Issue by jessicahawkins3344** [live-composer/live-composer-page-builder#472]
@lumberman - Please take a look if  #possible

## **UPDATED:** 
THE NEW JSON based code, when adding your own template files through either your functions.php or own plugin using the `dslc_get_templates` filter will correctly register the template, and save so that it is visible in LC Editor, however, when trying to use -- the HTML is NOT being rendered correctly .

## **Note:** 
If you simply 'import' the code, it renders fine. This is only seen when hard-coding new templates using the new JSON Based Code.
_____________________________________________________________________________________________________

The export page or section code 'syntax' or 'layout' has changed... for instance, sites with older lc installed export code example:

`[dslc_modules_section show_on="desktop tablet phone" type="wrapper" columns_spacing="spacing" bg_color="" bg_image_thumb="disabled" dslca-img-url="" bg_image="" bg_image_repeat="repeat" bg_image_position="left top" bg_image_attachment="scroll" bg_image_size="auto" bg_video="" bg_video_overlay_color="#000000" bg_video_overlay_opacity="0" border_color="" border_width="0" border_style="solid" border="top right bottom left" margin_h="0" margin_b="0" padding="80" padding_h="0" custom_class="" custom_id="" ] [dslc_modules_area last="yes" first="no" size="12"] [dslc_module]YTo0OntzOjE4OiJtb2R1bGVfaW5zdGFuY2VfaWQiO2k6MTA2O3M6NzoicG9zdF9pZCI7aToyO3M6MTE6ImRzbGNfbV9zaXplIjtzOjI6IjEyIjtzOjk6Im1vZHVsZV9pZCI7czoxNjoiRFNMQ19UZXh0X1NpbXBsZSI7fQ==[/dslc_module] [/dslc_modules_area] [/dslc_modules_section] `

New export page/section code example:

`[{"show_on":"desktop tablet phone","type":"wrapper","columns_spacing":"spacing","bg_color":"rgb(0,190,198)","bg_image_thumb":"disabled","dslca-img-url":"","bg_image":"","bg_image_repeat":"no-repeat","bg_image_position":"right top","bg_image_attachment":"fixed","bg_image_size":"auto","bg_video":"","bg_video_overlay_color":"rgb(0,190,198)","bg_video_overlay_opacity":"0","border_color":"","border_width":"0","border_style":"solid","border":"top right bottom left","margin_h":"0","margin_b":"0","padding":"80","padding_h":"12","custom_class":"","custom_id":"","element_type":"row","content":[{"element_type":"module_area","last":"yes","first":"no","size":"12","content":[{"css_show_on":"desktop tablet phone","content":"<h3 style=\\\"text-align: right;\\\">PAGE SUBHEADING</h3>\n<h1 style=\\\"text-align: right;\\\">Main Page Heading 4</h1>","css_custom":"enabled","css_min_height":"0","css_margin_top":"0","css_margin_right":"0","css_margin_bottom":"0","css_margin_left":"0","css_main_padding_vertical":"0","css_main_padding_horizontal":"10","css_main_bg_color":"rgba(0,0,0,0)","css_main_bg_img_repeat":"repeat","css_main_bg_img_attch":"scroll","css_main_bg_img_pos":"top left","css_main_border_color":"#ffffff","css_main_border_width":"5","css_main_border_trbl":"right ","css_main_border_radius_top":"0","css_main_border_radius_bottom":"0","css_main_color":"rgba(0,0,0,0)","css_main_font_size":"18","css_main_font_weight":"300","css_main_font_family":"Lato","css_main_font_style":"normal","css_main_letter_spacing":"0","css_main_line_height":"28","css_main_margin_bottom":"10","css_main_text_align":"left","css_h1_color":"rgb(255,255,255)","css_h1_font_size":"55","css_h1_font_weight":"200","css_h1_font_family":"Work Sans","css_h1_font_style":"normal","css_h1_letter_spacing":"0","css_h1_line_height":"55","css_h1_margin_bottom":"0","css_h1_padding_vertical":"0","css_h1_padding_horizontal":"0","css_h1_text_align":"left","css_h1_bg_color":"rgba(0,0,0,0)","css_h1_border_color":"rgba(0,0,0,0)","css_h1_border_width":"0","css_h1_border_trbl":"top right bottom left","css_h1_border_radius_top":"0","css_h1_border_radius_bottom":"0","css_h2_color":"rgb(11,200,207)","css_h2_font_size":"18","css_h2_font_weight":"300","css_h2_font_family":"Lato","css_h2_font_style":"normal","css_h2_letter_spacing":"0","css_h2_line_height":"20","css_h2_margin_bottom":"0","css_h2_padding_vertical":"0","css_h2_padding_horizontal":"0","css_h2_text_align":"left","css_h2_bg_color":"rgba(0,0,0,0)","css_h2_border_color":"rgba(0,0,0,0)","css_h2_border_width":"0","css_h2_border_trbl":"top right bottom left","css_h2_border_radius_top":"0","css_h2_border_radius_bottom":"0","css_h3_color":"rgb(51,51,51)","css_h3_font_size":"20","css_h3_font_weight":"700","css_h3_font_family":"Montserrat","css_h3_font_style":"normal","css_h3_letter_spacing":"5","css_h3_line_height":"22","css_h3_margin_bottom":"5","css_h3_padding_vertical":"0","css_h3_padding_horizontal":"0","css_h3_text_align":"left","css_h3_bg_color":"rgba(0,0,0,0)","css_h3_border_color":"rgba(0,0,0,0)","css_h3_border_width":"0","css_h3_border_trbl":"top right bottom left","css_h3_border_radius_top":"0","css_h3_border_radius_bottom":"0","css_h4_color":"rgba(0,0,0,0)","css_h4_font_size":"24","css_h4_font_weight":"300","css_h4_font_family":"Lato","css_h4_font_style":"normal","css_h4_letter_spacing":"0","css_h4_line_height":"34","css_h4_margin_bottom":"10","css_h4_padding_vertical":"0","css_h4_padding_horizontal":"0","css_h4_text_align":"left","css_h4_bg_color":"rgba(0,0,0,0)","css_h4_border_color":"rgba(0,0,0,0)","css_h4_border_width":"0","css_h4_border_trbl":"top right bottom left","css_h4_border_radius_top":"0","css_h4_border_radius_bottom":"0","css_h5_color":"rgba(0,0,0,0)","css_h5_font_size":"17","css_h5_font_weight":"400","css_h5_font_family":"Open Sans","css_h5_font_style":"normal","css_h5_letter_spacing":"0","css_h5_line_height":"27","css_h5_margin_bottom":"15","css_h5_padding_vertical":"0","css_h5_padding_horizontal":"0","css_h5_text_align":"left","css_h5_bg_color":"rgba(0,0,0,0)","css_h5_border_color":"rgba(0,0,0,0)","css_h5_border_width":"0","css_h5_border_trbl":"top right bottom left","css_h5_border_radius_top":"0","css_h5_border_radius_bottom":"0","css_h6_color":"rgba(0,0,0,0)","css_h6_font_size":"15","css_h6_font_weight":"400","css_h6_font_family":"Open Sans","css_h6_font_style":"normal","css_h6_letter_spacing":"0","css_h6_line_height":"25","css_h6_margin_bottom":"15","css_h6_padding_vertical":"0","css_h6_padding_horizontal":"0","css_h6_text_align":"left","css_h6_bg_color":"rgba(0,0,0,0)","css_h6_border_color":"rgba(0,0,0,0)","css_h6_border_width":"0","css_h6_border_trbl":"top right bottom left","css_h6_border_radius_top":"0","css_h6_border_radius_bottom":"0","css_link_color":"rgb(43, 218, 91)","css_link_color_hover":"rgb(39, 211, 86)","css_blockquote_link_color":"rgba(0,0,0,0)","css_blockquote_link_color_hover":"rgba(0,0,0,0)","css_li_color":"rgba(0,0,0,0)","css_li_font_size":"18","css_li_font_weight":"300","css_li_font_family":"Lato","css_li_line_height":"28","css_ul_margin_bottom":"25","css_ul_margin_left":"25","css_ul_style":"none","css_ol_style":"none","css_ul_li_margin_bottom":"10","css_li_bg_color":"rgba(0,0,0,0)","css_li_border_color":"rgba(0,0,0,0)","css_li_border_width":"0","css_li_border_trbl":"top right bottom left","css_li_border_radius_top":"0","css_li_border_radius_bottom":"0","css_li_padding_vertical":"0","css_li_padding_horizontal":"0","css_inputs_color":"#4d4d4d","css_inputs_font_size":"13","css_inputs_font_weight":"500","css_inputs_font_family":"Open Sans","css_inputs_line_height":"23","css_inputs_margin_bottom":"15","css_inputs_padding_vertical":"10","css_inputs_padding_horizontal":"15","css_inputs_bg_color":"#fff","css_inputs_border_color":"#ddd","css_inputs_border_width":"1","css_inputs_border_trbl":"top right bottom left","css_inputs_border_radius":"0","css_blockquote_color":"rgba(0,0,0,0)","css_blockquote_font_size":"13","css_blockquote_font_weight":"400","css_blockquote_font_family":"Open Sans","css_blockquote_line_height":"22","css_blockquote_margin_bottom":"0","css_blockquote_margin_left":"0","css_blockquote_padding_vertical":"0","css_blockquote_padding_horizontal":"0","css_blockquote_text_align":"left","css_blockquote_bg_color":"rgba(0,0,0,0)","css_blockquote_bg_img_repeat":"repeat","css_blockquote_bg_img_attch":"scroll","css_blockquote_bg_img_pos":"top left","css_blockquote_border_color":"rgba(0,0,0,0)","css_blockquote_border_width":"0","css_blockquote_border_trbl":"top right bottom left","css_blockquote_border_radius_top":"0","css_blockquote_border_radius_bottom":"0","css_button_bg_color":"#5890e5","css_button_bg_color_hover":"#5890e5","css_button_border_color":"#5890e5","css_button_border_color_hover":"#5890e5","css_button_border_width":"0","css_button_border_trbl":"top right bottom left","css_button_border_radius":"3","css_button_color":"#fff","css_button_color_hover":"#fff","css_button_font_size":"13","css_button_font_weight":"500","css_button_font_family":"Open Sans","css_button_line_height":"13","css_button_padding_vertical":"10","css_button_padding_horizontal":"15","css_res_t":"disabled","css_res_t_margin_bottom":"0","css_res_t_main_padding_vertical":"0","css_res_t_main_padding_horizontal":"0","css_res_t_main_font_size":"13","css_res_t_main_line_height":"22","css_res_t_main_text_align":"left","css_res_t_h1_font_size":"13","css_res_t_h1_line_height":"13","css_res_t_h1_margin_bottom":"15","css_res_t_h1_text_align":"left","css_res_t_h2_font_size":"13","css_res_t_h2_line_height":"13","css_res_t_h2_margin_bottom":"15","css_res_t_h2_text_align":"left","css_res_t_h3_font_size":"13","css_res_t_h3_line_height":"13","css_res_t_h3_margin_bottom":"15","css_res_t_h3_text_align":"left","css_res_t_h4_font_size":"13","css_res_t_h4_line_height":"13","css_res_t_h4_margin_bottom":"15","css_res_t_h4_text_align":"left","css_res_t_h5_font_size":"13","css_res_t_h5_line_height":"13","css_res_t_h5_margin_bottom":"15","css_res_t_h5_text_align":"left","css_res_t_h6_font_size":"13","css_res_t_h6_line_height":"13","css_res_t_h6_margin_bottom":"15","css_res_t_h6_text_align":"left","css_res_t_li_font_size":"13","css_res_t_li_line_height":"22","css_res_t_ul_margin_bottom":"25","css_res_t_ul_margin_left":"25","css_res_t_ul_li_margin_bottom":"10","css_res_t_li_padding_vertical":"0","css_res_t_li_padding_horizontal":"0","css_res_t_blockquote_font_size":"13","css_res_t_blockquote_line_height":"22","css_res_t_blockquote_margin_bottom":"0","css_res_t_blockquote_margin_left":"0","css_res_t_blockquote_padding_vertical":"0","css_res_t_blockquote_padding_horizontal":"0","css_res_t_blockquote_text_align":"left","css_res_p":"disabled","css_res_ph_margin_bottom":"0","css_res_ph_main_padding_vertical":"0","css_res_ph_main_padding_horizontal":"0","css_res_ph_main_font_size":"13","css_res_ph_main_line_height":"22","css_res_p_main_text_align":"left","css_res_ph_h1_font_size":"13","css_res_ph_h1_line_height":"13","css_res_ph_h1_margin_bottom":"15","css_res_p_h1_text_align":"left","css_res_ph_h2_font_size":"13","css_res_ph_h2_line_height":"13","css_res_p_h2_margin_bottom":"15","css_res_p_h2_text_align":"left","css_res_ph_h3_font_size":"13","css_res_ph_h3_line_height":"13","css_res_p_h3_margin_bottom":"15","css_res_p_h3_text_align":"left","css_res_ph_h4_font_size":"13","css_res_ph_h4_line_height":"13","css_res_p_h4_margin_bottom":"15","css_res_p_h4_text_align":"left","css_res_ph_h5_font_size":"13","css_res_ph_h5_line_height":"13","css_res_p_h5_margin_bottom":"15","css_res_p_h5_text_align":"left","css_res_ph_h6_font_size":"13","css_res_ph_h6_line_height":"13","css_res_p_h6_margin_bottom":"15","css_res_p_h6_text_align":"left","css_res_p_li_font_size":"13","css_res_p_li_line_height":"22","css_res_p_ul_margin_bottom":"25","css_res_p_ul_margin_left":"25","css_res_p_ul_li_margin_bottom":"10","css_res_p_li_padding_vertical":"0","css_res_p_li_padding_horizontal":"0","css_res_p_blockquote_font_size":"13","css_res_p_blockquote_line_height":"22","css_res_p_blockquote_margin_bottom":"0","css_res_p_blockquote_margin_left":"0","css_res_p_blockquote_padding_vertical":"0","css_res_p_blockquote_padding_horizontal":"0","css_res_p_blockquote_text_align":"left","css_anim":"none","css_anim_delay":"0","css_anim_duration":"650","css_anim_easing":"ease","css_load_preset":"none","module_instance_id":"7668670926b","post_id":2935,"dslc_m_size":"12","module_id":"DSLC_Text_Simple","element_type":"module","last":"yes","dslc_m_size_last":"yes","module_render_nonajax":true}]}]}]`

For whatever reason, this new export code is not always importing correctly, especially when I add custom lc templates into my theme... here is screencastify:

https://drive.google.com/file/d/0B23NdAgM5vWJLUtqV00tWmFablE/view

**Comment by xdima92x** [comment]
Hi @jessicahawkins3344 I checked your code of JSON. You have a problem with [the double slash](https://www.evernote.com/shard/s554/sh/5bf2825d-22f2-413e-880c-9f0b0fc90152/dcad7a58df77acdb). You need to delete it. Please check this branch - https://github.com/live-composer/live-composer-page-builder/tree/issue/472 I improved the LC.

**Comment by xdima92x** [comment]
Hi @lumberman Please check this branch - https://github.com/live-composer/live-composer-page-builder/tree/issue/472

**Comment by jessicahawkins3344** [comment]
@xdima92x @lumberman - xdima, that commit from that branch in the ajax.php file worked with NO problems by changing line 429. This should be pushed to the current LC version asap
